xtra-libp2p/src/lib.rs

```diff
@@ -19,7 +19,7 @@ use libp2p_core::Negotiated;
 use libp2p_core::PeerId;
 
 pub mod dialer;
-mod endpoint;
+pub mod endpoint;
```

Comment:
Is that actually necessary? How did we avoid that before our changes? :D

Reply:
it selectively exported `Endpoint` struct before.